flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java

```diff
@@ -222,15 +223,25 @@ protected void registerNamespace(
                             Collections.singletonList(simplifiedRexNode),
                             reducedNodes);
             // check whether period is the unsupported expression
-            if (!(reducedNodes.get(0) instanceof RexLiteral)) {
-                throw new UnsupportedOperationException(
+            final RexNode reducedNode = reducedNodes.get(0);
+            if (!(reducedNode instanceof RexLiteral)) {
+                throw new ValidationException(
                         String.format(
                                 "Unsupported time travel expression: %s for the expression can not be reduced to a constant by Flink.",
                                 periodNode));
             }
 
-            RexLiteral rexLiteral = (RexLiteral) (reducedNodes).get(0);
-            TimestampString timestampString = rexLiteral.getValueAs(TimestampString.class);
+            final SqlTypeName sqlTypeName = ((RexLiteral) reducedNode).getTypeName();
+            if (!(sqlTypeName == SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE
+                    || sqlTypeName == SqlTypeName.TIMESTAMP)) {
+                throw newValidationError(
+                        periodNode,
+                        Static.RESOURCE.illegalExpressionForTemporal(sqlTypeName.getName()));
+            }
+
+            RexLiteral rexLiteral = (RexLiteral) reducedNode;
+            TimestampString timestampString =
+                    ((RexLiteral) reducedNode).getValueAs(TimestampString.class);
```

Comment:
Use rexLiteral here

Reply:
ah, I wanted to delete the line with `rexLiteral` 🤦